Step3-인증을 통한 기능 구현

cyr9210 opened the pull request:
안녕하세요.
인수 테스트 기반 TDD 3단계 인증을 통한 기능 구현 리뷰 요청드립니다.
감사합니다.

cyr9210 committed:
refactor: 2단계 피드백 반영

cyr9210 committed:
test: 토큰 발급 기능 인수테스트 작성

cyr9210 committed:
test: 내 정보 기능 인수 테스트 작성

cyr9210 committed:
fix: 내 정보 기능 완성

- @AuthenticationPrincipal 추가

cyr9210 committed:
test: header() -> auth().oauth2(token)

cyr9210 committed:
test: 즐겨찾기 기능 인수테스트 작성

cyr9210 committed:
feat: 즐겨찾기 기능 추가

- 엔티티 정의
- 추가
- 목록조회
- 삭제

wooobo commented on src/main/java/nextstep/subway/favorite/application/FavoriteService.java:
```diff
@@ -0,0 +1,61 @@
+@Service
+public class FavoriteService {
+
+    private MemberRepository memberRepository;
```
`final`을 붙여주세요!

wooobo commented on src/main/java/nextstep/subway/favorite/exception/FavoriteNotFoundException.java:
```diff
@@ -0,0 +1,10 @@
+    private static final String FAVORITE_CAN_NOT_FIND = "favorite can not find.";
+
+    public FavoriteNotFoundException() {
+        super(FAVORITE_CAN_NOT_FIND);
```
계속해서 하나의 예외메시지만을 가지는 커스텀 예외가 증가되고있습니다.
관리가 힘들어지는 포인트가 되지 않을까요?

wooobo commented on src/test/java/nextstep/subway/auth/acceptance/AuthAcceptanceTest.java:
```diff
@@ -1,24 +1,94 @@
+        로그인_완료됨(response);
     }
 
     @DisplayName("Bearer Auth 로그인 실패")
```
비밀번호 일치하지 않아 실패한다는 문구가 들어가지면 더 좋지 않을까요?

wooobo commented on src/test/java/nextstep/subway/favorite/FavoriteAcceptanceTest.java:
```diff
@@ -1,8 +1,119 @@
+    @Test
+    void manageFavorite() {
+        // when
+        ExtractableResponse<Response> createResponse = 즐겨찾기_생성_요청(token, 강남역.getId(), 서초역.getId());
```
`즐겨찾기_생성됨` 과 `즐겨찾기_삭제됨` 하나의 테스트로 통합하신 이유가 있으실까요?

cyr9210 replied:
특별한 의도가 있었던것은 아닙니다.
미리 작성되어 있던 회원정보관리 테스트 포맷을 보고 이런식으로 작성해야하나? 라고 생각되어 작성했었습니다.
기능별 분리 및 예외테스트 추가하였습니다.!

wooobo reviewed (changes requested):
안녕하세요 3단계 잘 진행해주셨네요!

몇가지 코멘트 남겨드렸는데 확인후 재요청 부탁드려요 🙏

cyr9210 committed:
refactor: 피드백 반영

- final 추가
- notFoundException 통일 EntityNotFoundException
- test 설명 구체화
- FavoriteAcceptanceTest 테스트 각 기능 별 분리
   - 예외 테스트 추가

wooobo commented on src/main/java/nextstep/subway/path/application/PathFinder.java:
```diff
@@ -50,7 +50,7 @@ private GraphPath getGraphPath(Station source, Station target) {
             graphPath = shortestPath.getPath(source, target);
         } catch (IllegalArgumentException e) {
-            throw new StationNotFoundException();
+            throw new EntityNotFoundException();
```
`throw new EntityNotFoundException(ErrorCode::STATION_NOT_FOUND));`

커스텀 `예외`를 만들었다는건, 특별한 뭔가를 전달해줄수 있지 않을까요?
그렇지 않다면, 자바 기본 `Exception`을 사용해도 크게 다르지 않을것같아요.

cyr9210 replied:
음.. 커스텀예외를 만들어 쓰지 않았는데요
말씀하신 피드백 방향으로도 작업을 해보도록 하겠습니다.

wooobo reviewed (approved):
피드백 고민해보셨군요 :)

3단계도 잘진행해주셨습니다.
코멘트 하나 남겨드렸어요~ 확인부탁드리고,
`merge`진행하겠습니다.
